synfig-studio/src/gui/states/state_bone.cpp

```diff
@@ -619,11 +619,7 @@ StateBone_Context::event_mouse_release_handler(const Smach::event& x)
 						list_node=ValueNode_StaticList::Handle::cast_dynamic(list_desc.get_value_node());
 						ValueDesc value_desc= ValueDesc(list_node,active_bone,list_desc);
 						setActiveBone->set_param("active_bone_node",value_desc.get_value_node());
-						if(get_work_area()->get_active_bone_value_node()){
-							setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
-						}else{
-							setActiveBone->set_param("prev_active_bone_node",ValueNode::Handle::cast_dynamic(ValueNode_Bone::create(Bone())));
-						}
+						setActiveBone->set_param("prev_active_bone_node",get_work_area()->get_active_bone_value_node());
```

Comment:
Shouldn't action `ValueNodeSetActiveBone` set was the previous one automatically?

Reply:
Actually, as I was unable to access anything related to rendering i.e, `WorkArea`, `CanvasView` from actions, I decided that the user will provide it. :sweat_smile: